test/framework/src/main/java/org/opensearch/test/OpenSearchIntegTestCase.java

```diff
@@ -2349,4 +2358,88 @@ protected ClusterState getClusterState() {
         return client(internalCluster().getClusterManagerName()).admin().cluster().prepareState().get().getState();
     }
 
+    /**
+     * Refreshes all indices in the cluster when forceRefresh set to true and
+     * checks if replica shards caught up with primary shard when Segment Replication is enabled.
+     */
+    protected void waitForReplication(boolean forceRefresh, String... indices) {
+        if (forceRefresh) {
+            refresh(indices);
+        }
+        if (isInternalCluster()) {
```

Comment:
Majority of Integration tests in server module this should not be a issue as they all use InternalTestCluster class, but for REST tests in other modules this might be an issue. Example test is using ExternalTestCluster ([example scenario](https://build.ci.opensearch.org/job/gradle-check/32469/testReport/junit/org.opensearch.index.mapper/TokenCountFieldMapperIntegrationIT/testSearchByTokenCount__storeCountedFields_true_loadCountedFields_true_/))

Reply:
The risk here is that if a test that uses an external cluster tries to use "wait for replication" with a segment replication-enabled index, then it will just do nothing and not actually wait for replication

Reply:
what you said is right, but the main reason for adding boolean check for externalClusters is:
for core logic waitForReplicasToCatchUpWithPrimary() to work we need instance of IndexShard to check the store contents of primary and replica. To get instance of any class we usually do something like internalCluster().getInstance(IndicesService.class, node)  on InternalTestCluster. But for ExternalTestCluster we don't have this.
the one option I can think of to resolve it:
-> For external cluster tests we can  use a bit different logic. Instead of checking store contents of primary and replica, we make an API call of (cat/segrep or indexstats) to see if replica is behind primary.
Then there will be two different logics for internal and external cluster. The only drawback of using these api's is it will be slow compared to doing store check.
What's your opinion with this approach?